pkg/tsdb/sqleng/sql_engine_test.go

```diff
@@ -9,317 +9,308 @@ import (
+		t.Run("interpolate $__interval in $__timeGroup", func(t *testing.T) {
+			sql, err := Interpolate(query, timeRange, "select $__timeGroupAlias(time,$__interval)")
+			require.NoError(t, err)
+			require.Equal(t, "select $__timeGroupAlias(time,1m)", sql)
+		})
 
-				So(sql, ShouldEqual, "select 60000 ")
-			})
+		t.Run("interpolate $__interval_ms", func(t *testing.T) {
+			sql, err := Interpolate(query, timeRange, "select $__interval_ms ")
+			require.NoError(t, err)
+			require.Equal(t, "select 60000 ", sql)
+		})
 
-			Convey("interpolate __unixEpochFrom function", func() {
-				sql, err := Interpolate(query, timeRange, "select $__unixEpochFrom()")
-				So(err, ShouldBeNil)
+		t.Run("interpolate __unixEpochFrom function", func(t *testing.T) {
+			sql, err := Interpolate(query, timeRange, "select $__unixEpochFrom()")
+			require.NoError(t, err)
+			require.Equal(t, fmt.Sprintf("select %d", from.Unix()), sql)
+		})
 
-				So(sql, ShouldEqual, fmt.Sprintf("select %d", from.Unix()))
-			})
+		t.Run("interpolate __unixEpochTo function", func(t *testing.T) {
+			sql, err := Interpolate(query, timeRange, "select $__unixEpochTo()")
+			require.NoError(t, err)
+			require.Equal(t, fmt.Sprintf("select %d", to.Unix()), sql)
+		})
+	})
 
-			Convey("interpolate __unixEpochTo function", func() {
-				sql, err := Interpolate(query, timeRange, "select $__unixEpochTo()")
-				So(err, ShouldBeNil)
+	t.Run("Given row values with time.Time as time columns", func(t *testing.T) {
+		var nilPointer *time.Time
+
+		fixtures := make([]interface{}, 5)
+		fixtures[0] = dt
+		fixtures[1] = &dt
+		fixtures[2] = earlyDt
+		fixtures[3] = &earlyDt
+		fixtures[4] = nilPointer
+
+		for i := range fixtures {
+			ConvertSqlTimeColumnToEpochMs(fixtures, i)
+		}
+
+		t.Run("When converting them should return epoch time with millisecond precision", func(t *testing.T) {
+			expected := float64(dt.UnixNano()) / float64(time.Millisecond)
+			expectedEarly := float64(earlyDt.UnixNano()) / float64(time.Millisecond)
+
+			require.Equal(t, expected, fixtures[0].(float64))
+			require.Equal(t, expected, fixtures[1].(float64))
+			require.Equal(t, expectedEarly, fixtures[2].(float64))
+			require.Equal(t, expectedEarly, fixtures[3].(float64))
+			require.Nil(t, fixtures[4])
+		})
+	})
 
-				So(sql, ShouldEqual, fmt.Sprintf("select %d", to.Unix()))
-			})
+	t.Run("Given row values with int64 as time columns", func(t *testing.T) {
+		tSeconds := dt.Unix()
+		tMilliseconds := dt.UnixNano() / 1e6
+		tNanoSeconds := dt.UnixNano()
+		var nilPointer *int64
+
+		fixtures := make([]interface{}, 7)
+		fixtures[0] = tSeconds
+		fixtures[1] = &tSeconds
+		fixtures[2] = tMilliseconds
+		fixtures[3] = &tMilliseconds
+		fixtures[4] = tNanoSeconds
+		fixtures[5] = &tNanoSeconds
+		fixtures[6] = nilPointer
+
+		for i := range fixtures {
+			ConvertSqlTimeColumnToEpochMs(fixtures, i)
+		}
+
+		t.Run("When converting them should return epoch time with millisecond precision", func(t *testing.T) {
```

Comment:
Do you mind just removing t.Run here? Every t.Run introduces a sub-test, and I don't think it makes sense to group assertions as sub-tests. IMO a sub-test should correspond to a test case.